### ISSUE TITLE
PyPy fails SimpleUtilityTestCase.test_resolveIPv6

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7731]
|<img alt="amauryfa's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| amauryfa reported|
|-|-|
|Trac ID|trac#7731|
|Type|enhancement|
|Created|2014-12-03 12:21:11Z|
|Branch|https://github.com/twisted/twisted/tree/resolveipv6-pypy-7731|

For ipv6 addresses, pypy's getaddrinfo() returns 'flowinfo' and 'scope_id' as long objects instead of int. This causes a test to fail.

```
Traceback (most recent call last):
  File "/var/lib/buildbot/bot-glyph-1/trusty64-pypy-2.4/Twisted/twisted/internet/test/test_tcp.py", line 2516, in test_resolveIPv6
    self.assertIsInstance(result[2], int) # flow info
  File "/var/lib/buildbot/bot-glyph-1/trusty64-pypy-2.4/Twisted/twisted/trial/_synctest.py", line 664, in assertIsInstance
    instance, classOrTuple, suffix))
twisted.trial.unittest.FailTest: 0L is not an instance of <type 'int'>
```

* either relax the test a bit, and use `self.assertIsInstance(result[2], (int, long))`

* or fix PyPy. 

The relevant code in PyPy is in `pypy/module_socket/interp_socket.py`: the call to `space.wrap(addr.get_flowinfo())` returns a long, because `get_flowinfo()` returns a `lltype.Unsigned` (in `rpython/rlib/rsocket.py`).

A possible fix in PyPy is to change `space.wrap()` (in `pypy/objspace/std/objspace.py`) to call `space.newint()` in more cases.

<details><summary>Searchable metadata</summary>

```
trac-id__7731 7731
type__enhancement enhancement
reporter__amauryfa amauryfa
priority__normal normal
milestone__PyPy_support PyPy-support
branch__branches_resolveipv6_pypy_7731 branches/resolveipv6-pypy-7731
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1417609271528212 1417609271528212
changetime__1441595286490423 1441595286490423
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

This might make sense to fix in PyPy as well but I think the proposed Twisted test change is also pretty reasonable.  It's generally not safe to assume some arbitrary API will represent integer results as one of `int` or `long` specifically - and in most cases these types are interchangeable anyway.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [45640]) Branching to resolveipv6-pypy-7731.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

Looks good locally, all supported builders green, pypy builder is running here: https://buildbot.twistedmatrix.com/builders/ubuntu15.04-pypy25/builds/37

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

PyPy builder doesn't have the failure. Woo!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Since the tests were changed, maybe we need to add a comment in the test and explain that cpython 2 uses int while pypy 2 uses long.

other than that it looks good.

Please check my comment and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

Hi Adi, what returns int and what returns long on Python 2 is interesting -- for example, system calls on Windows sometimes return longs, and 32/64 can sometimes mean a long is produced on 32 and an int on 64. So, we should allow both in most cases -- it's not just this one case.

Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45647]) Merge resolveipv6-pypy-7731: Make the IPv6 tests pass on PyPy

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7731](https://github.com/twisted/twisted/issues/7731)
